src/main/java/com/foundation/search/model/Criteria.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2018 Jalasoft.
+ * 2643 Av Melchor Perez de Olguin, Colquiri Sud, Cochabamba, Bolivia.
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information of
+ * Jalasoft, ("Confidential Information").  You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jalasoft.
+ */
+
+package com.foundation.search.model;
+
+public class Criteria {
+    public String path;
```

Comment:
Is there some especial reason to have the parameters as "public"?
if not, I think you should consider to make them private or at least protected and refactor the methods to return the proper parameters